### PR TITLE
high-flow commit 2

### PR DESCRIPTION
**Pull request by kasProg**
### What was done:

1. Fixed seed in Dropout function
2. Introduced clear option of using subset-data to train in StreamflowExample-DI.py

### Verification/Results:

1. Fixing seed did not change the NSE much, but helped in reproducing results
2. Adding flag of subset_train, lets other users easily modify the training data.

### CODE REVIEW

**Comment by taddyb** [comment]
Following best practices, we shouldn't be setting random seeds in functions. Please pass it into the `createMask()` as a variable so that we don't lose track of it

**Comment by taddyb** [comment]
Writing up @kamleshcodes changes here so they aren't lost.

functions changed:
- `CudnnLstm()`
- `createMask()`

what was done:
- `CudnnLstm()` now takes an optional seed argument and sets a seed to be added to `createMask()`

``` py
class CudnnLstm(nn.Module):
    def __init__(self, *, inputSize, hiddenSize, dr=0.5, drMethod="drW", gpu=0, seed=42):
```

``` py
createMask(self.w_hh, self.dr, self.seed)
```

from my understanding, the seed is used to fix the dropout.
